phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -5480,21 +5522,21 @@ private boolean evaluateStmtProperties(MetaProperties metaProperties, MetaProper
             }
         }
 
-        if (metaProperties.getPhoenixTTL() != null) {
+        if (metaProperties.getTTL() != null) {
             if (table.getType() == PTableType.INDEX) {
                 throw new SQLExceptionInfo.Builder(
                         SQLExceptionCode.CANNOT_SET_OR_ALTER_PROPERTY_FOR_INDEX)
                         .build()
                         .buildException();
             }
-            if (table.getType() != PTableType.TABLE && table.getType() != SYSTEM) {
+            if (table.getType() != PTableType.TABLE) {
```

Comment:
TTL is not supported for SYSTEM tables as well.

Reply:
Yes, for SYSTEM table TTL is not supported at Phoenix level as discussed I am not sure why this change is shown by git as this is already there :- https://github.com/apache/phoenix/blame/40a30ac66a7655003f52e816525c3e209d44a083/phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java#L2065